providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -533,11 +535,13 @@ static int dh_gen_common_set_params(void *genctx, const OSSL_PARAM params[])
     p = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_FFC_TYPE);
     if (p != NULL) {
         if (p->data_type != OSSL_PARAM_UTF8_STRING
-            || ((gctx->gen_type =
+            || ((gen_type =
                  dh_gen_type_name2id_w_default(p->data, gctx->dh_type)) == -1)) {
             ERR_raise(ERR_LIB_PROV, ERR_R_PASSED_INVALID_ARGUMENT);
             return 0;
         }
+        if (gen_type != -1)
+            gctx->gen_type = gen_type;
```

Comment:
I think you could drop the if-statement and just assign to  `gctx->gen_type` here.  At this point, we know that `gen_type != -1`.

Reply:
Is that true?  If p->data_type == OSSL_PARAM_UTF8_string, gen_type will not get reassigned in the if_conditional above.  Or am I missing something?

Reply:
You are right, Neil.  Sorry for the distraction.

Reply:
No worries, thanks for looking, I thought I had missed something